biz.aQute.launchpad/src/aQute/launchpad/LaunchpadBuilder.java

```diff
@@ -45,13 +46,18 @@
 
 		//
 		// We only want the raw setup and not the run spec since this
-		// makes it impossible to start a clean framework
+		// makes it impossible to start a clean framework. Make them
+		// immutable so that they can't be modified accidentally.
 		//
 
-		projectTestSetup.runbundles.clear();
-		projectTestSetup.runpath.clear();
-		projectTestSetup.properties.clear();
-		projectTestSetup.runfw.clear();
+		projectTestSetup.runbundles = Collections.EMPTY_LIST;
```

Comment:
Please use `Collections.emptyList()` and `Collections.emptyMap()` to avoid warnings in Eclipse.

Reply:
Odd, I didn't get a warning and I'm using Eclipse 2018-12.